src/libraries/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/Rfc2898DeriveBytes.OneShot.cs

```diff
@@ -0,0 +1,134 @@
+
+            return Pbkdf2DeriveBytes(password.AsSpan(), new ReadOnlySpan<byte>(salt), iterations, length, hashAlgorithm);
+        }
+
+        public static byte[] Pbkdf2DeriveBytes(
+            ReadOnlySpan<char> password,
+            ReadOnlySpan<byte> salt,
+            int iterations,
+            int length,
+            HashAlgorithmName hashAlgorithm)
+        {
+            if (length < OneShotMinExtractLength)
+                throw new ArgumentOutOfRangeException(nameof(length), SR.Format(SR.Argument_Rfc2898_MinLength, OneShotMinExtractLength));
+
+            byte[] result = new byte[length];
+            Pbkdf2DeriveBytes(password, salt, iterations, hashAlgorithm, result);
+            return result;
+        }
+
+        public static void Pbkdf2DeriveBytes(
+            ReadOnlySpan<char> password,
+            ReadOnlySpan<byte> salt,
+            int iterations,
+            HashAlgorithmName hashAlgorithm,
+            Span<byte> destination)
+        {
+            byte[] passwordBytes = CryptoPool.Rent(Encoding.UTF8.GetMaxByteCount(password.Length));
+            int passwordBytesWritten = Encoding.UTF8.GetBytes(password, passwordBytes);
+
+            Pbkdf2DeriveBytesCore(passwordBytes, salt, iterations, hashAlgorithm, destination);
+            CryptoPool.Return(passwordBytes, clearSize: passwordBytesWritten);
+        }
+
+        private static void Pbkdf2DeriveBytesCore(
+            ReadOnlySpan<byte> password,
+            ReadOnlySpan<byte> salt,
+            int iterations,
+            HashAlgorithmName hashAlgorithm,
+            Span<byte> destination)
+        {
+            if (salt.Length < OneShotMinSaltLength)
+                throw new ArgumentOutOfRangeException(nameof(salt), SR.Format(SR.Argument_Rfc2898_SaltTooShort, OneShotMinSaltLength));
+            if (iterations < OneShotMinIterations)
+                throw new ArgumentOutOfRangeException(nameof(iterations), SR.Format(SR.Argument_Rfc2898_MinIterations, OneShotMinIterations));
+            if (destination.Length < OneShotMinExtractLength)
+                throw new ArgumentOutOfRangeException(nameof(destination), SR.Format(SR.Argument_Rfc2898_MinLength, OneShotMinExtractLength));
+            if (string.IsNullOrEmpty(hashAlgorithm.Name))
+                throw new ArgumentException(SR.Cryptography_HashAlgorithmNameNullOrEmpty, nameof(hashAlgorithm));
+
+            string hashAlgorithmName = hashAlgorithm.Name;
+
+            // MD5 intentionally left out.
+            if (hashAlgorithmName != HashAlgorithmName.SHA1.Name &&
+                hashAlgorithmName != HashAlgorithmName.SHA256.Name &&
+                hashAlgorithmName != HashAlgorithmName.SHA384.Name &&
+                hashAlgorithmName != HashAlgorithmName.SHA512.Name)
+            {
+                throw new CryptographicException(SR.Format(SR.Cryptography_UnknownHashAlgorithm, hashAlgorithmName));
+            }
+
+            if (destination.IsEmpty)
```

Comment:
This shouldn't be reachable, given the check against OneShotMinExtractLength, right?  An assert might be better if you're just trying to avoid passing NULL to the native layers.

Reply:
There is no limit on the extract length anymore, so I think this should be retained. The implementations assert `destination` is not empty now.